src/platform/backends/backend_utils/backend_utils.cpp

```diff
@@ -107,3 +108,25 @@ std::string mp::backend::image_format_for(const mp::Path& image_path)
 
     return image_record["format"].toString().toStdString();
 }
+
+void mp::backend::shutdown_instance(const std::string& vm_name, VirtualMachine::State& state,
+                                    QTimer& delay_shutdown_timer, std::chrono::minutes delay,
+                                    std::function<void()> const& process_shutdown_request)
+{
```

Comment:
This should be check if the state is already stopped.

Otherwise, you end up being able to change the state of stopped instances into DELAYED_SHUTDOWN which will never happen as they are stopped already.

Reply:
Yep, you're right:)